datafusion/catalog/src/information_schema.rs

```diff
@@ -403,73 +404,89 @@ impl InformationSchemaConfig {
                 let arg_types = arg_types
                     .into_iter()
-                    .map(|t| t.to_string())
+                    .map(|t| remove_native_type_prefix(NativeType::from(t)))
                     .collect::<Vec<_>>();
                 (arg_types, return_type)
             })
-            .collect::<Vec<_>>())
+            .collect::<BTreeSet<_>>())
     }
 }
 
 fn get_udaf_args_and_return_types(
     udaf: &Arc<AggregateUDF>,
-) -> Result<Vec<(Vec<String>, Option<String>)>> {
+) -> Result<BTreeSet<(Vec<String>, Option<String>)>> {
     let signature = udaf.signature();
     let arg_types = signature.type_signature.get_example_types();
     if arg_types.is_empty() {
-        Ok(vec![(vec![], None)])
+        Ok(vec![(vec![], None)].into_iter().collect::<BTreeSet<_>>())
     } else {
         Ok(arg_types
             .into_iter()
             .map(|arg_types| {
                 // only handle the function which implemented [`ScalarUDFImpl::return_type`] method
-                let return_type =
-                    udaf.return_type(&arg_types).ok().map(|t| t.to_string());
+                let return_type = udaf
+                    .return_type(&arg_types)
+                    .ok()
+                    .map(|t| remove_native_type_prefix(NativeType::from(t)));
                 let arg_types = arg_types
                     .into_iter()
-                    .map(|t| t.to_string())
+                    .map(|t| remove_native_type_prefix(NativeType::from(t)))
                     .collect::<Vec<_>>();
                 (arg_types, return_type)
             })
-            .collect::<Vec<_>>())
+            .collect::<BTreeSet<_>>())
     }
 }
 
 fn get_udwf_args_and_return_types(
     udwf: &Arc<WindowUDF>,
-) -> Result<Vec<(Vec<String>, Option<String>)>> {
+) -> Result<BTreeSet<(Vec<String>, Option<String>)>> {
     let signature = udwf.signature();
     let arg_types = signature.type_signature.get_example_types();
     if arg_types.is_empty() {
-        Ok(vec![(vec![], None)])
+        Ok(vec![(vec![], None)].into_iter().collect::<BTreeSet<_>>())
     } else {
         Ok(arg_types
             .into_iter()
             .map(|arg_types| {
                 // only handle the function which implemented [`ScalarUDFImpl::return_type`] method
                 let arg_types = arg_types
                     .into_iter()
-                    .map(|t| t.to_string())
+                    .map(|t| remove_native_type_prefix(NativeType::from(t)))
                     .collect::<Vec<_>>();
                 (arg_types, None)
             })
-            .collect::<Vec<_>>())
+            .collect::<BTreeSet<_>>())
     }
 }
 
+#[inline]
+fn remove_native_type_prefix(native_type: NativeType) -> String {
+    // native_type.to_string() is like "NATIVE_TYPE_PREFIX + native_type"
+    // here is safe to unwrap directly
+    native_type
+        .to_string()
+        .strip_prefix(NATIVE_TYPE_PREFIX)
+        .unwrap()
+        .to_string()
```

Comment:
It would be nice to avoid having to do string manipulations on something we know is valid at compile time

My reading suggests that this is equivalent and is simpler, and would save a few copies and allocations

```suggestion
    format!("{native_type:?}")
```